File: src/main/java/com/garageos/modules/vehicle/service/VehicleService.java
package com.garageos.modules.vehicle.service;

import com.garageos.core.enums.vehicle.RcVerificationStatus;
import com.garageos.modules.vehicle.dto.request.CreateVehicleRequest;
import com.garageos.modules.vehicle.dto.response.VehicleResponse;
import org.springframework.data.domain.Page;

public interface VehicleService {

    VehicleResponse createVehicle(CreateVehicleRequest request);

    VehicleResponse getVehicle(Long id);

    Page<VehicleResponse> getAllVehicles(
            int page,
            int size,
            String sortBy,
            String direction);

    VehicleResponse updateVehicle(
            Long id,
            CreateVehicleRequest request);

    void deleteVehicle(Long id);

    VehicleResponse getVehicleByRegistrationNumber(
            String registrationNumber);

    /**
     * Mission backlog #1 — RC verification. Role-restricted at the
     * controller (Manager/Service Advisor/Owner); records who verified it
     * and when.
     */
    VehicleResponse setRcVerification(
            Long id,
            RcVerificationStatus status,
            String documentReference);
}

File: src/main/resources/static/services/vehicleService.js
window.VehicleService = {

    async createVehicle(request) {

        return await Api.post(
            "/vehicles",
            request
        );

    },

    async updateVehicle(id, request) {

        return await Api.put(
            "/vehicles/" + id,
            request
        );

    },

    async searchByRegistrationNumber(registrationNumber) {

        return await Api.get(
            "/vehicles/search?registrationNumber=" +
            encodeURIComponent(registrationNumber)
        );

    }

};